lab_1/main.py

```diff
@@ -2,34 +2,51 @@
 Lab 1
 Language detection
 """
-
-
+import re
 def tokenize(text: str) -> list or None:
     """
     Splits a text into tokens, converts the tokens into lowercase,
     removes punctuation and other symbols from words
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    if isinstance(text, str):
+        text = text.lower()
+        for i in range(len(text)):
+            text = re.sub('[^a-züöäß \n]','', text) #ищем слова(все то, что не цифры и не прочие знаки)
+        text = text.split()
+        return text #иначе будет выводиться None
     pass
 
-
 def remove_stop_words(tokens: list, stop_words: list) -> list or None:
     """
     Removes stop words
     :param tokens: a list of tokens
     :param stop_words: a list of stop words
     :return: a list of tokens without stop words
     """
+    if isinstance(tokens, list) and isinstance(stop_words, list):
+        if tokens:
+            tokens = [n for n in tokens if n not in stop_words]
+            return tokens
     pass
 
-
 def calculate_frequencies(tokens: list) -> dict or None:
     """
     Calculates frequencies of given tokens
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
+    if isinstance(tokens, list):
```

Comment:
There should be a check for the input arguments at the top of the function - after the documentation string. Remove nesting here and extract the check into a separate `if` block earlier.